src/main/webapp/open_hours.js

```diff
@@ -45,36 +50,51 @@ class TimeOption {
     this.dayLabel.classList.add("day-label");
     this.dayOptionArea.appendChild(this.dayLabel);
 
-    this.dayOpenLabel = document.createElement("label");
-    this.dayOpenLabel.textContent = "Open";
-    this.dayOpenLabel.classList.add("day-open-label");
-    this.dayOptionArea.appendChild(this.dayOpenLabel);
-    this.dayOpenInput = document.createElement("input");
-    this.dayOpenInput.setAttribute("type", "radio");
-    this.dayOpenInput.setAttribute("name", `${this.day}-isOpen`);
-    this.dayOpenInput.setAttribute("value", "open");
-    this.dayOptionArea.appendChild(this.dayOpenInput);
-
-    this.dayClosedLabel = document.createElement("label");
-    this.dayClosedLabel.textContent = "Closed";
-    this.dayClosedLabel.classList.add("day-closed-label");
-    this.dayOptionArea.appendChild(this.dayClosedLabel);
-    this.dayClosedInput = document.createElement("input");
-    this.dayClosedInput.setAttribute("type", "radio");
-    this.dayClosedInput.setAttribute("name", `${this.day}-isOpen`);
-    this.dayClosedInput.setAttribute("value", "closed");
-    this.dayOptionArea.appendChild(this.dayClosedInput);
-
-    this.radioElements = [this.dayOpenInput, this.dayClosedInput];
-    if (this.forRegistration) {
-      this.dayOpenInput.setAttribute("checked", "checked");
-      this.timeInputArea.classList.add("show-time-area");
-    } else if (!this.organizationDay.propertyMap.isOpen) {
-      this.dayClosedInput.setAttribute("checked", "checked");
-      this.timeInputArea.classList.add("hide-time-area");
-    } else {
-      this.dayOpenInput.setAttribute("checked", "checked");
-      this.timeInputArea.classList.add("show-time-area");
+
+    // Only show open closed options for organization modification
+    if (!document.getElementById("event-registration-form-area")) {
```

Comment:
Generally special casing if a certain id is available is a bad code smell.

You have several options here; if you want to do so: 
   move this code into a :   renderIntoElement(element) {}  function, that will allow you decide to create the elements in the calling code. You can also extract a new class that does this instead that encapsulates the data as well as creates the elements.

Reply:
Let's continue to think about this:

I believe we talked about the options of : extract creating these open/close options in a new function (and call it elsewhere!) or pass in a constructor argument that tells this object whether or not to create these elements.